ckanext/gla/plugin.py

```diff
@@ -1,26 +1,34 @@
 from collections import OrderedDict
-from typing import Any
+from typing import Any, Optional, NoReturn
 
+import logging
 import ckan.plugins as plugins
 import ckan.plugins.toolkit as toolkit
-from ckan.common import _
+from ckan.common import _, g, request
+from ckan import model
+from ckan.model import User
+from ckan.lib import captcha, signals
 from ckan.config.declaration import Declaration, Key
-from ckan.lib.helpers import markdown_extract, ungettext, dict_list_reduce
+from ckan.lib.helpers import markdown_extract, ungettext, dict_list_reduce, flash_error
 from ckan.types import Schema, Validator
+from ckan.views.user import PerformResetView
+from flask import abort as flask_abort
 from markupsafe import Markup
 
-from . import auth, custom_fields, helpers, search, timestamps, views
+from . import auth, custom_fields, helpers, search, timestamps, views, user
 from .search_highlight import action, query
 
 TABLE_FORMATS = toolkit.config.get("ckan.harvesters.table_formats").split(" ")
 REPORT_FORMATS = toolkit.config.get("ckan.harvesters.report_formats").split(" ")
 GEOSPATIAL_FORMATS = toolkit.config.get("ckan.harvesters.geospatial_formats").split(" ")
 
+log = logging.getLogger(__name__)
 
 class GlaPlugin(plugins.SingletonPlugin, toolkit.DefaultDatasetForm):
     plugins.implements(plugins.IConfigurer)
     plugins.implements(plugins.IConfigDeclaration)
     plugins.implements(plugins.IAuthFunctions, inherit=True)
+    plugins.implements(plugins.IAuthenticator)
```

Comment:
If we add `inherit=True` here, you won't have to implement empty methods, only the ones we want to overwrite.